src/hooks/useInitReefState.ts

```diff
@@ -1,57 +1,128 @@
+  }
+  setIsLoading(false);
+},[provider]);
+  return [accounts,isLoading];
+};
 
-export interface State {
-  loading: boolean;
-  provider?: Provider;
-  network?: Network;
-  error?: any; // TODO!
-  extension?: InjectedExtension;
+export const reefAccountToReefSigner = (accountsFromUtilLib:any,injectedSigner:InjectedSigner)=>{
+  const resultObj = {
+    name:'reef',
+    sig:injectedSigner,
+  };
+  let reefSigners = <any[]>[];
+  for(let i = 0;i<accountsFromUtilLib?.data.length;i++){
+    let reefAccount = accountsFromUtilLib.data[i].data;
+    let toReefSigner = {
+      name:reefAccount.name,
+      address:reefAccount.address,
+      source:reefAccount.source,
+      genesisHash:reefAccount.genesisHash,
+    };
+    reefSigners.push(toReefSigner);
+  }
+  resultObj['accounts'] = reefSigners;
+  return resultObj;
 }
 
 export const useInitReefState = (
   applicationDisplayName: string,
-  options: {network?: Network; ipfsHashResolverFn?: ipfsUrlResolverFn;} = {},
+  options: StateOptions = {},
 ): State => {
+  const {
+    explorerClient, dexClient, ipfsHashResolverFn,
+  } = options;
   const [accounts, extension, loadingExtension, errExtension] = useInjectExtension(applicationDisplayName);
-  const { network: selectedNetwork, provider } = useObservableState(selectedNetworkProvider$);
-  const isProviderLoading = useObservableState(providerConnState$.pipe(map((v) => !v.isConnected)), false);
-
-  const [loading, setLoading] = useState(false);
-
+  const jsonAccounts = { accounts, injectedSigner: extension?.signer };
+  const selectedNetwork: Network|undefined = useObservableState(reefState.selectedNetwork$);
+  const provider = useObservableState(reefState.selectedProvider$) as Provider|undefined;
+  const [loading, setLoading] = useState(true);
   useEffect(() => {
+    const network = getNetworkFallback();
+
     if (!accounts || !accounts.length || !extension) {
       return;
     }
 
     const jsonAccounts = { accounts, injectedSigner: extension?.signer };
-    initReefState({
-      network: options.network,
-      jsonAccounts,
-      ipfsHashResolverFn: options.ipfsHashResolverFn,
-    });
-  }, [accounts, extension, options]);
+      reefState.initReefState({
+        network,
+        jsonAccounts,
+        ipfsHashResolverFn,
+      });
+  }, [accounts, extension]);
+
+
+  useEffect(() => {
+    setNftIpfsResolverFn(ipfsHashResolverFn);
+  }, [ipfsHashResolverFn]);
 
   useEffect(() => {
-    setLoading(loadingExtension && isProviderLoading);
-  }, [isProviderLoading, loadingExtension]);
+    initAxiosClients(selectedNetwork, explorerClient, dexClient);
+  }, [selectedNetwork, explorerClient, dexClient]);
+
+  let accountsFromUtilLib: any= useObservableState(reefState.accounts$);
+
+  const [loadedReefSigners,isLoadingReefSigners] = getReefSignersArray([reefAccountToReefSigner(accountsFromUtilLib,jsonAccounts.injectedSigner!)],provider!);
+  useEffect(() => {
+          accountsSubj.next(loadedReefSigners || []);
```

Comment:
accounts in react-lib should be transformed from util-lib observable and not set as Subject - like:
export const injectedSigners$ = reefState.accounts$.pipe(...)
and should be referenced where signers are needed